week19/restock-shopping-cart_starter/cart.jsx

```diff
@@ -53,47 +44,16 @@ const useDataApi = (initialUrl, initialData) => {
   }, [url]);
 
 
-  return [state, setUrl];
-};
-
-const dataFetchReducer = (state, action) => {
-  switch (action.type) {
-    case "FETCH_INIT":
-      return {
-        ...state,
-        isLoading: true,
-        isError: false,
-      };
-    case "FETCH_SUCCESS":
-      return {
-        ...state,
-        isLoading: false,
-        isError: false,
-        data: action.payload,
-      };
-    case "FETCH_FAILURE":
-      return {
-        ...state,
-        isLoading: false,
-        isError: true,
-      };
-    default:
-      throw new Error();
-  }
+  return [inventory, setUrl];
 };
 
 // /component/Products.jsx
 // import useDataApi ...
-const Products = (props) => {
+const Products = () => {
   const [items, setItems] = useState([]);
   const [cart, setCart] = useState([]);
   const [query, setQuery] = useState("http://localhost:1337/api/products");
-  const [{ data }, apiCall] = useDataApi(
-    "http://localhost:1337/api/products",
-    {
-      data: [],
-    }
-  );
+  const [inventory, doFetch] = useDataApi(query);
```

Comment:
inventory = the inventory state variable 
doFetch = the function to set the url state variable